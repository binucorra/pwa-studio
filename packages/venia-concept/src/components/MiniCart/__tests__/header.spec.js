import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import Header from '../header';

jest.mock('../trigger');
jest.mock('src/components/Icon', () => ({
    __esModule: true,
    default: () => <span>Mock Icon Component</span>
}));

test('it renders correctly', () => {
    const props = {
        isEditingItem: false
    };

    const tree = createTestInstance(<Header {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('it changes the title when editing an item', () => {
    const props = {
        isEditingItem: true
    };

    const tree = createTestInstance(<Header {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});
