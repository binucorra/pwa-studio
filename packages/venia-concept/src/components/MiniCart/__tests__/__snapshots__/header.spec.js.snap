// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it changes the title when editing an item 1`] = `
<div>
  <h2>
    Edit Cart Item
  </h2>
  ( Trigger Component Here )
</div>
`;

exports[`it renders correctly 1`] = `
<div>
  <h2>
    Shopping Cart
  </h2>
  ( Trigger Component Here )
</div>
`;
